pkg/cri/server/container_create_windows_test.go

```diff
@@ -245,3 +250,179 @@ func TestHostProcessRequirements(t *testing.T) {
+	testPid := uint32(1234)
+	nsPath := "test-cni"
+	c := newTestCRIService()
+	ImgEntrypoint := []string{`"C:\My Folder\MyProcess.exe" -arg1 "test value"`}
+	ImgCmd := []string{`cmd -args "hello world"`}
+
+	expectedCommandLine := `"C:\My Folder\MyProcess.exe" -arg1 "test value" "cmd -args \"hello world\""`
+
+	containerConfig, sandboxConfig, imageConfig := getCreateContainerTestDataForArgsEscapedTests(ImgEntrypoint, ImgCmd, nil, nil)
+	runtimeSpec, err := c.containerSpec(testID, testSandboxID, testPid, nsPath, testContainerName, testImageName, containerConfig, sandboxConfig, imageConfig, nil, config.Runtime{})
+	assert.NoError(t, err)
+	assert.NotNil(t, runtimeSpec)
+
+	checkRuntimeSpec(t, runtimeSpec, nil, expectedCommandLine)
+}
+
+// override image entrypoint and cmd in shell form with container args and verify expected runtime spec
+func TestShellFormImgEntrypointCmdWithCtrArgs(t *testing.T) {
+	testID := "test-id"
+	testSandboxID := "sandbox-id"
+	testContainerName := "container-name"
+	testPid := uint32(1234)
+	nsPath := "test-cni"
+	c := newTestCRIService()
+	ImgEntrypoint := []string{`"C:\My Folder\MyProcess.exe" -arg1 "test value"`}
+	ImgCmd := []string{`cmd -args "hello world"`}
+	args := []string{`cmd -args "additional args"`}
+
+	expectedCommandLine := `"C:\My Folder\MyProcess.exe" -arg1 "test value" "cmd -args \"additional args\""`
+
+	containerConfig, sandboxConfig, imageConfig := getCreateContainerTestDataForArgsEscapedTests(ImgEntrypoint, ImgCmd, nil, args)
+	runtimeSpec, err := c.containerSpec(testID, testSandboxID, testPid, nsPath, testContainerName, testImageName, containerConfig, sandboxConfig, imageConfig, nil, config.Runtime{})
+	assert.NoError(t, err)
+	assert.NotNil(t, runtimeSpec)
+
+	checkRuntimeSpec(t, runtimeSpec, nil, expectedCommandLine)
+}
+
+// check image entrypoint and cmd in exec form without overriding with container command and args and verify expected runtime spec
+func TestExecFormImgEntrypointCmdWithoutCtrArgs(t *testing.T) {
+	testID := "test-id"
+	testSandboxID := "sandbox-id"
+	testContainerName := "container-name"
+	testPid := uint32(1234)
+	nsPath := "test-cni"
+	c := newTestCRIService()
+	imgEntrypoint := []string{`C:\My Folder\MyProcess.exe`, "-arg1", "test value"}
+	imgCmd := []string{"cmd", "-args", "hello world"}
+
+	expectedArgs := []string{`C:\My Folder\MyProcess.exe`, "-arg1", "test value", "cmd", "-args", "hello world"}
+
+	containerConfig, sandboxConfig, imageConfig := getCreateContainerTestDataForArgsEscapedTests(imgEntrypoint, imgCmd, nil, nil)
+	runtimeSpec, err := c.containerSpec(testID, testSandboxID, testPid, nsPath, testContainerName, testImageName, containerConfig, sandboxConfig, imageConfig, nil, config.Runtime{})
+	assert.NoError(t, err)
+	assert.NotNil(t, runtimeSpec)
+
+	checkRuntimeSpec(t, runtimeSpec, expectedArgs, "")
+}
+
+// override image cmd by container args in exec form and verify expected runtime spec
+func TestExecFormImgEntrypointCmdWithCtrArgs(t *testing.T) {
```

Comment:
I was trying to think of how to simplify these names, but that led to thinking that these should probably just be a single test where these are testcases and use t.Run(). So TestArgsEscaped (or any name you think makes sense here) and then something like https://github.com/containerd/containerd/blob/main/pkg/cri/sbserver/container_stats_list_linux_test.go#L63-L89 (arbitrarily chosen, there's a ton of t.Run() examples to choose from here)